### PR TITLE
feat: modularize console implementation for late UEFI output

### PR DESCRIPTION
**Pull request by mkroening**
This is a larger restructuring, allowing UEFI to use native console implementations as fallback once boot services have exited. This should also allow nice GOP support later.

This is in preparation for https://github.com/hermit-os/loader/pull/285 (modulo GOP support).

### CODE REVIEW

**Comment by mkroening** [comment]
CC: @sarahspberrypi, @jounathaen

File: src/arch/aarch64/console.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use core::ptr::NonNull;
+
+use hermit_dtb::Dtb;
+
+pub struct Console {
+	stdout: NonNull<u8>,
+}
+
+fn stdout() -> u32 {
```
Comment (jounathaen) [inline src/arch/aarch64/console.rs:9]: I found this fn name a little confusing. Maybe something like `get_stdout` or `stdout_addr` would be more self-explanatory. A doc-comment might also help.

File: src/os/uefi/console.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	BootServices,
+	Native { console: arch::Console },
+}
+
+impl Console {
+	const fn new() -> Self {
+		Self::None
+	}
+
+	fn exit_boot_services(&mut self) {
```
Comment (jounathaen) [inline src/os/uefi/console.rs:24]: Do I get this correct: This is a callback for exiting the boot services? If so, I'd suggest modifying the name to reflect this. Also, I'm a huge fan of doc comments also for private fns :wink:

Comment (mkroening) [reply src/os/uefi/console.rs:24]: Correct. I was inspired by https://github.com/rust-osdev/uefi-rs/blob/137932086eff0e0a7d0e5a63426a9e47f32439b3/uefi-services/src/lib.rs#L179.

For me, the few lines of code made the purpose self-explanatory, though we can always add docs, if people are confused.

**Pull request merged**